core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/mysql/MysqlDatabaseService.java

```diff
@@ -18,16 +30,39 @@ public synchronized static MysqlDatabaseService getInstance() {
         return INSTANCE;
     }
 
-    private MysqlDatabaseService() {}
-
     @Override
     public MysqlDatabase getDatabase(Connection connection) {
-        return null;
+        String userName = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, USER_KEY);
+        String userPassword = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, PASSWORD_KEY);
+        Optional<String> schemaName = DatabaseHandler.getInstance().getOptionalParameterWithKey(connection, SCHEMA_KEY);
```

Comment:
shouldn't this be renamed to database?

Reply:
shouldn't it be a mandatory parameter? Can you connect (via jdbc) to a MySQL database without defining the database parameter?